### DISCUSSION TITLE
Choose Language (Auto-Detect) Option

### DISCUSSION THREAD

**Discussion by Joelson-Forte** [SubtitleEdit/subtitleedit#8633] [Ideas]
Hi @niksedk 

Please add "Auto-Detect" option in choose language. This option is useful when there is more than one language to be transcribed.

![2024-07-13_18h52_42](https://github.com/user-attachments/assets/2495203a-802a-4d35-a8a7-b45ae14975cd)

**Comment by Joelson-Forte** [comment]
@niksedk 
Do you think it is a good idea to add this option? The biggest advantage is that it will be possible to transcribe multiple clips with different languages ​​without having to choose the language for each clip.

![xxx](https://github.com/user-attachments/assets/3f9c5c21-855c-4d98-bfb0-ba1269d0a4e1)

**Comment by niksedk** [comment]
Auto-detect from what? Filename or content?
Also read this: https://github.com/SYSTRAN/faster-whisper/issues/918

I do not think this is a good idea for now... 
I have seen Whisper recognize the audio, even if the language changes (probably requires non EN models)

**Comment by Joelson-Forte** [comment]
Hi @niksedk 

Here's an example:
I added three clips in Batch mode:
Clip 1 is in English
Clip 2 is in German
Clip 3 is in Portuguese

![2024-08-05_09h31_51](https://github.com/user-attachments/assets/1a29d166-30fa-4110-acf7-7bda1e53e647)





But I can only choose one language to transcribe in "Choose Language". That's the problem.

Maybe this can be solved if the "--language None" command can be enabled. From what I understand, if the '--language None" command is entered in the command line, Whisper will use the first 30 seconds of each audio to detect the language of each file automatically. But unfortunately this command does not currently work in Subtitle Edit.
